src/v/cluster/rm_stm.h

```diff
@@ -131,6 +143,25 @@ class rm_stm final : public persisted_stm {
         std::vector<abort_index> abort_indexes;
         model::offset offset;
         std::vector<seq_entry> seqs;
+
+        struct tx_seqs_snapshot {
+            model::producer_identity pid;
+            model::tx_seq tx_seq;
+        };
+
+        struct inflight_snapshot {
+            model::producer_identity pid;
+            int64_t counter;
+        };
+
+        struct expiration_snapshot {
+            model::producer_identity pid;
+            expiration_info info;
```

Comment:
it doesn't compile, instead of storing the whole expiration_info just store the timeout